app/src/main/java/org/groundplatform/android/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -197,23 +197,24 @@ internal constructor(
   fun getTaskViewModel(taskId: String): AbstractTaskViewModel? {
     val viewModels = taskViewModels.value
 
-    val task = tasks.first { it.id == taskId }
-
     if (viewModels.containsKey(taskId)) {
       return viewModels[taskId]
     }
 
-    return try {
-      val viewModel = viewModelFactory.create(getViewModelClass(task.type))
-      taskViewModels.value[task.id] = viewModel
-
-      val taskData: TaskData? = if (shouldLoadFromDraft) getValueFromDraft(task) else null
-      viewModel.initialize(job, task, taskData)
-      viewModel
-    } catch (e: Exception) {
-      Timber.e("ignoring task with invalid type: $task.type")
-      null
-    }
+    return tasks
+      .firstOrNull { it.id == taskId }
```

Comment:
Why would we displaying the data collection interface if there are no tasks, or why would we be requesting a taskId which doesn't existing? Is there a race condition somewhere?